federatedscope/cl/dataloader/Cifar10.py

```diff
@@ -0,0 +1,187 @@
+import math
+import torch
+import torch.nn as nn
+import torch.nn.functional as F
+import torch.backends.cudnn as cudnn
+import torchvision
+import torchvision.transforms as T
+import torchvision.transforms.functional as TF
+from torch.utils.data import DataLoader, Dataset
+from torchvision.datasets import CIFAR10, CIFAR100
+import pickle as pkl
+import numpy as np
+
+
+
+class SimCLRTransform():
+    def __init__(self, is_sup, image_size=32):
+        self.transform = T.Compose([
+            T.RandomResizedCrop(image_size, scale=(0.5, 1.0), interpolation=T.InterpolationMode.BICUBIC),
+            T.RandomHorizontalFlip(p=0.5),
+            T.RandomApply([T.ColorJitter(0.4,0.4,0.2,0.1)], p=0.8),
+            T.RandomGrayscale(p=0.2),
+            T.RandomApply([T.GaussianBlur(kernel_size=3, sigma=(0.1, 2.0))], p=0.5),
+            T.ToTensor(),
+            T.Normalize((0.5, 0.5, 0.5), (0.5, 0.5, 0.5))
+        ])
+
+        self.mode = is_sup
+
+    def __call__(self, x):
+        if(self.mode):
+            return self.transform(x)
+        else:
+            x1 = self.transform(x)
+            x2 = self.transform(x)
+            return x1, x2 
+
+def Cifar4CL(config):
+    
+    transform_train = SimCLRTransform(is_sup=False, image_size=32)
+    transform_test = T.Compose([
+        T.ToTensor(), 
+        T.Normalize((0.5, 0.5, 0.5), (0.5, 0.5, 0.5))]
```

Comment:
is it conventional to use 0.5 rather than the sample mean?

Reply:
Setting both parameters to 0.5 and using with T.totensor() can force the data to be scaled to the [-1,1] interval

Reply:
T.totensor() can take the value of image from 0-255 to 0-1, and T.Normalize(0.5,0.5) can take 0-1 to -1-1 using function (x-mean)/std